security/s2a/internal/crypter/aesgcm_test.go

```diff
@@ -0,0 +1,319 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package crypter
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+)
+
+// Constants indicating whether the test vector is valid or not.
+const (
+	validResult   = "valid"
+	invalidResult = "invalid"
+)
+
+// cryptoTestVector is a struct for a AES-GCM test vector
+type cryptoTestVector struct {
```

Comment:
Sorry I forgot to mention this earlier. We also need the NIST and IEEE test vectors (see [here](https://github.com/grpc/grpc-java/blob/62620ccd00b4bc772ba6622a7c1b192dea45747c/alts/src/test/java/io/grpc/alts/internal/AesGcmHkdfAeadCrypterTest.java#L156)). There are just a few of them, so they should be reasonable to hardcode as test vectors.